core/src/main/scala/org/bitcoins/core/protocol/ln/LnTaggedFields.scala

```diff
@@ -141,46 +139,40 @@ object LnTaggedFields {
 
           loop(newRemaining, fields.:+(tag))
 
-        case _ :: _ | _ :: _ :: _ =>
-          throw new IllegalArgumentException("Failed to parse LnTaggedFields, needs 15bits of meta data to be able to parse")
         case Nil =>
           fields
+        case _ =>
+          throw new IllegalArgumentException("Failed to parse LnTaggedFields, needs 15bits of meta data to be able to parse")
       }
     }
 
-    val tags = loop(u5s.toList, Vector.empty)
-
-    val paymentHashTag = {
 
-      val fOpt = tags.find(_.isInstanceOf[LnTag.PaymentHashTag])
-      if (fOpt.isDefined) {
-        fOpt.get.asInstanceOf[LnTag.PaymentHashTag]
-      } else {
-        throw new IllegalArgumentException(s"Payment hash must be defined in a LnInvoice")
-      }
+    val tags = loop(u5s.toList, Vector.empty)
 
+    def getTag[T <: LnTag : ClassTag]: Option[T] = {
```

Comment:
private

Reply:
I don't think that should matter here? `getTag` is defined inside `fromUInt5s`, so shouldn't be accessible from the outside anyway. Same as with `loop` a few lines up